pkg/sharing/secret_exports.go

```diff
@@ -0,0 +1,98 @@
+package sharing
+
+import (
+	"sync"
+
+	"github.com/go-logr/logr"
+	sgv1alpha1 "github.com/vmware-tanzu/carvel-secretgen-controller/pkg/apis/secretgen/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type SecretExports struct {
+	log logr.Logger
+
+	exportedSecretsLock sync.RWMutex
+	exportedSecrets     map[string]exportedSecret
+}
+
+type exportedSecret struct {
+	*sgv1alpha1.SecretExport
+	*corev1.Secret
+}
+
+func (p exportedSecret) Key() string {
+	if p.SecretExport.Namespace == "" {
+		panic("Internal inconsistency: missing namespace")
```

Comment:
should we debate `panic` v.s. errors ?  I think this is getting called in the reconcile loop and do we really want to panic during runtime or only during initialization?  If there is a busted resource is there a way/reason for us to skip it and continue to the other(s) vs. bailing entirely?